src/imagej/convert.py

```diff
@@ -562,6 +563,30 @@ def _rename_xarray_dims(xarr, new_dims: Sequence[str]):
     return xarr.rename(dim_map)
 
 
+def _create_imagej_metadata(
+    axes: Sequence[Union["jc.DefaultLinearAxis", "jc.EnumeratedAxis"]],
+    dim_seq: Sequence[str],
+) -> dict:
+    """
+    Create the ImageJ metadata attribute dictionary for xarray's global attributes.
+    """
+    ij_metadata = {}
+    assert len(axes) == len(dim_seq)
```

Comment:
I would put this check within an `if` clause, and raise a `ValueError` if they are not the same.

`assert`s in general are widely considered bad practice outside of tests, although I don't have a good authority to reference on that. At the very least, I'd prefer the `ValueError` over the `AssertionError` you'll get now, as the `ValueError` is more specific about what is wrong.

Reply:
Agreed a4ea186440bae6f6dcaa258b926f32942422c5a1!